src/applications/disability-benefits/2346/Components/VeteranInformationPage.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { getVeteranInformationData } from '../actions';
+
+class VeteranInformationPage extends React.Component {
+  componentDidMount() {
+    this.props.getVeteranInformationData();
+  }
+  render() {
+    const { first, last, gender, dateOfBirth } = this.props;
+    return (
+      <div>
+        <p>This is the personal information we have for you.</p>
+        <div>
+          <div className="usa-alert schemaform-sip-alert">
+            <div className="usa-alert-body">
+              <p className="vads-u-margin--1px">
+                {first} {last}
+              </p>
+              <p className="vads-u-margin--1px">Date of Birth: {dateOfBirth}</p>
+              <p className="vads-u-margin--1px"> Gender: {gender}</p>
+            </div>
+          </div>
+          <br />
+        </div>
+      </div>
+    );
+  }
+}
+
+VeteranInformationPage.propTypes = {
+  first: PropTypes.string.isRequired,
+  last: PropTypes.string.isRequired,
+  gender: PropTypes.string.isRequired,
+  dateOfBirth: PropTypes.string.isRequired,
+  getVeteranInformationData: PropTypes.func.isRequired,
+};
+
+const mapStateToProps = state => ({
+  first: state.form2346Reducer?.formData?.veteranFullName?.first,
```

Comment:
Missing default props or initialState in reducer

Reply:
@bradl3yC The `initialState` is set in `src/applications/disability-benefits/2346/reducers/form2346Reducer.js` on line 9 `formData: {},`

Reply:
@mojaray2k That is an empty object.  All of those keys you marked as required do not exist initially on that object.  When this component loads, and that object is empty, what happens to these props that have an undefined value?

Reply:
@bradl3yC The data for each of those props are loaded via the `getVeteranInformationData` dispatch call. I know that I need a loading indicator so that when the data is being requested to show the user there is a delay.

Reply:
@mojaray2k I understand it will have data in it _at some point_ but initially its empty.  When that component is mounted, there is the potential it will be missing data.  This will lead to a console warning regarding the missing props.  Its a simple fix to add defaultProps with empty strings so the component behaves as it should.  You've done this for a previous component already.

Reply:
@bradl3yC Am I correctly addressing your concerns if not please provide an example?

Reply:
Oh ok @bradl3yC I see what your are saying doing it now.

Reply:
@bradl3yC Done